arch/common/CMakeLists.txt

```diff
@@ -39,3 +38,29 @@ if (DEFINED CONFIG_ARM OR DEFINED CONFIG_X86
     OR DEFINED CONFIG_SOC_OPENISA_RV32M1_RISCV32)
   zephyr_linker_sources(ROM_START SORT_KEY 0x0 rom_start_offset.ld)
 endif()
+
+
+# isr_tables is a normal CMake library and not a zephyr_library because it
+# should not be --whole-archive'd
+if (CONFIG_GEN_ISR_TABLES)
+  add_library(isr_tables
+    isr_tables.c
+  )
+
+  set_target_properties(
+    isr_tables
+    PROPERTIES
+    COMPILE_DEFINITIONS
+    __ZEPHYR_SUPERVISOR__
+    )
+
+  target_include_directories(isr_tables PRIVATE
+    ${ZEPHYR_BASE}/kernel/include
+    ${ZEPHYR_BASE}/arch/${ARCH}/include
+    )
+
+  add_dependencies(isr_tables zephyr_generated_headers)
+  target_link_libraries(isr_tables zephyr_interface)
+
+  set_property(GLOBAL APPEND PROPERTY ZEPHYR_NO_WHOLE_ARCHIVE_LIBS isr_tables)
```

Comment:
This should not be needed.
When using a normal CMake library that is not `--whole-archive`'d, then we should be using normal CMake library dependencies.

That would be, the libraries that is dependent upon `isr_tables` should have a line, similar to:
```
zephyr_library_link_libraries_ifdef(CONFIG_GEN_ISR_TABLES isr_tables)
```

As example, this would work for arm targets:
```
diff --git a/arch/arm/core/aarch32/CMakeLists.txt b/arch/arm/core/aarch32/CMakeLists.txt
index 4d2898970e..d419b5fbf3 100644
--- a/arch/arm/core/aarch32/CMakeLists.txt
+++ b/arch/arm/core/aarch32/CMakeLists.txt
@@ -34,3 +34,4 @@ add_subdirectory_ifdef(CONFIG_ARM_NONSECURE_FIRMWARE cortex_m/tz)
 add_subdirectory_ifdef(CONFIG_CPU_CORTEX_R cortex_a_r)
 
 zephyr_linker_sources(ROM_START SORT_KEY 0x0vectors vector_table.ld)
+zephyr_library_link_libraries_ifdef(CONFIG_GEN_ISR_TABLES isr_tables)
```

and thus remove the need for the new `ZEPHYR_NO_WHOLE_ARCHIVE_LIBS` property.